qbraid/runtime/native/job.py

```diff
@@ -85,11 +84,29 @@ def cancel(self) -> None:
 
         self.client.cancel_job(self.id)
 
-    def result(self) -> "qbraid.runtime.GateModelJobResult":
+    def _build_runtime_gate_model_results(self, job_data, **kwargs) -> list[ExperimentalResult]:
+        """Build and return the results of a gate-model simulation.
+
+        Args:
+            job_data (dict): The job data dictionary.
+        """
+        measurement_counts = job_data.get("measurementCounts", {})
+        time_stamps: dict[str, Any] = job_data.get("timeStamps", {})
+        execution_duration: int = time_stamps.get("executionDuration", -1)
+        return [
+            ExperimentalResult(
+                state_counts=measurement_counts,
+                measurements=ResultFormatter.counts_to_measurements(measurement_counts),
```

Comment:
he explicit measurements are only needed if we don't have histogram counts data. If we only have the measurements array and no counts data, then we use the measurements to create the histogram counts. After we have the counts data, the full measurements array is no longer needed. If provided, we can still store it. But in my mind it doesn't make a lot of sense to go out of our way to "recreate" them from the counts if not provided